angular-workspace/projects/ni/nimble-angular/table-column/nimble-table-column-base.directive.ts

```diff
@@ -72,4 +72,12 @@ export class NimbleTableColumnBaseDirective<T extends TableColumn> {
     }
 
     public constructor(protected readonly renderer: Renderer2, protected readonly elementRef: ElementRef<T>) {}
+
+    public checkValidity(): boolean {
+        return this.elementRef.nativeElement.checkValidity();
+    }
+
+    public get validity(): TableColumnValidity {
```

Comment:
I looked briefly at it, trying to see what it would take, but I didn't get anywhere. I have a feeling that would be at odds with the implementation we're currently using (based on `src/utilities/models/tracker.ts`). I'm also not sure how likely it is that client code would be accessing individual flags in that object, vs dumping the object to the console for debugging, for example. What do you think?

Reply:
I was wondering the same thing about how likely it is that someone would actually write code that uses the output of that function vs using it in debugging. I mostly asked because depending on how we went about it, it might be easier to implement now rather than after we have a bunch of column types. A few ideas that immediately come to mind are:
- Export the Validity type for each column. This isn't a great experience, but it would at least allow someone to do something like `(myDateColumn.validity as DateTextColumnValidity).invalidCustomFormatters`
- Make `get validity()` abstract on the base directive so that each column has to implement it to return their specific validity type. I'm not 100% sure if TypeScript would make that typing easy or if we'd have to jump through any hoops to get that to work.

I don't have a problem waiting until we have a better use-case for needing improved typing before we come up with a plan. I just wanted to make sure we were explicitly making the decision not to do anything now.

Reply:
I would vote for giving it some run time to see if there's a need for improved typing. I have a feeling we won't need it.